libarchive-cph-static-feedstock/recipe/meta.yaml

```diff
@@ -0,0 +1,88 @@
+{% set version = "3.3.3" %}
+
+# Recipe for a static libarchive with support for the features needed for
+# conda-package-handling (bzip2, zlib, zstd).
+# This is not a general purpose libarchive
+package:
+  name: libarchive-cph-static
+  version: {{ version }}
+
+source:
+  url: https://github.com/libarchive/libarchive/releases/download/v{{ version }}/libarchive-{{ version }}.tar.gz
+  sha256: ba7eb1781c9fbbae178c4c6bad1c6eb08edab9a1496c64833d1715d022b30e2e
+  patches:
+    - patches/0001-Add-lib-to-CMAKE_FIND_LIBRARY_PREFIXES-for-lzma.patch
+    # fixed in 3.4.0
+    # - patches/0002-xar-fix-bad-seek-in-move_reading_point.patch
+    - patches/0003-VC9-compatibility-test-for-BCryptDeriveKeyPBKDF2.patch
+    - patches/0004-VC9-compatibility-define-BCRYPT_SUCCESS.patch
+    # This should not be applied in general. It exists so I can build it for
+    # debugging Visual Studio where warnings are errors for some reason.
+    # Hiding warnings is poor practice.
+    # - patches/0005-VC9-compatibility-ignore-some-warnings.patch
+    - patches/0006-VC-openssl-define-inline-as-__inline.patch
+    # These warnings are errors when compiling lz-4 support
+    - patches/0007-CMake-VC9-compatibility-ignore-some-warnings.patch
+    - patches/0008-VC9-compatibility-remove-C99.patch
+    - patches/0009-CMake-Force-Multi-threaded-DLL-runtime.patch
+    # 3.4.0 added official support for symlinks on win
+    # - patches/0010-Use-CreateSymbolicLinkW-on-Windows-for-symlinks-WIP.patch
+
+build:
+  number: 0
+  # skip: true  # [win and vc<14]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    # Does not work in this recips at present
+    # we manually convert C99 to C89 instead.
+    # - c99-to-c89  # [win and vc<14]
+    - msinttypes  # [win and vc<14]
+    - autoconf    # [unix]
+    - automake    # [unix]
+    - libtool     # [unix]
+    - pkg-config  # [unix]
+    - cmake       # [win]
+    - make        # [unix]
+    - ninja       # [win]
+    - m2-patch    # [win]
+  host:
+    - msinttypes  # [win and vc<14]
+    - bzip2
+    - libiconv    # [osx]
+    - zlib
+    - zstd
+
+test:
+  files:
+    - test/*
+  commands:
+    # Verify pkg-config file is in place.
+    - test -f "${PREFIX}/lib/pkgconfig/libarchive.pc"  # [unix]
+
+    # Verify headers are in place.
+    - test -f "${PREFIX}/include/archive.h"            # [unix]
+    - test -f "${PREFIX}/include/archive_entry.h"      # [unix]
+
+    # Verify libraries are in place.
+    - test -f "${PREFIX}/lib/libarchive.a"             # [unix]
+    - test -f "${PREFIX}/lib/libarchive_and_deps.a"    # [unix]
+
+    # Check for commands
+    - bsdtar --version | grep libzstd  # [unix]
+    - bsdtar --version                 # [not unix]
+    - bsdtar -tf test/imagesize-1.1.0-py37_0.tar.bz2
+    - bsdtar -tf test/imagesize-1.1.0-py37_0.conda
```

Comment:
if bsdtar is in these packages, will it conflict with the general purpose libarchive package?

Reply:
Also: how can we test it here if we can't have bsdtar in this package?

Reply:
Yes this will conflict with bsdtar in the libarchive package.  An simple solution would be to re-name the binary in this recipe to something unique (bsdtar_do_not_use_for_cph?).  

A short C program could be written to test out the functionality provided by the library.  Alternatively, the cph tests suite could be used.